### PR TITLE
mgmt, upgrade authorization to 2020-10-01

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
Live test passes.

- Clean MSGraph lib, as new m4 has bad performance on deduplication and results in lots of `AutoGenerated` classes.
- RoleAssignment is actually on 2020-10-01-preview, to avoid breaking change on the `condition` method.

Manual change:
- https://github.com/Azure/azure-sdk-for-java/pull/28334/commits/189b44c38700892520be56e12729cb3ba6fe9b64
- https://github.com/Azure/azure-sdk-for-java/pull/28334/commits/47cce03824a89c994d8ac24abd1b58188181fac1

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by XiaofeiCao — APPROVED** [review]
LGTM.

**Comment by picpromusic** [comment]
When this change will be released? The old API-Version 2020-08-01-preview which is used in 2.14.0 is disable in Azure. So all code that uses this library is potentially broken.

**Comment by weidongxu-microsoft** [comment]
@picpromusic 2.15.0 is to be released in May.

I don't think 2020-08-01-preview is to be deprecated soon. I've not heard any notification.

And even older preview version should still work, say 2018-01-01-preview.

**Comment by picpromusic** [comment]
@weidongxu-microsoft . It is already actually disabled in west europe. We get the following error: 
Status code 400, "{"error":{"code":"NoRegisteredProviderFound","message":"No registered resource provider found for location 'westeurope' and API version '2020-08-01-preview' for type 'roleDefinitions'. The supported api-versions are '2014-04-01-preview, 2014-07-01-preview, 2014-10-01-preview, 2015-05-01-preview, 2015-06-01, 2015-07-01, 2016-07-01, 2017-05-01, 2017-09-01, 2018-01-01-preview, 2018-07-01'. The supported locations are ''."}}"

**Comment by weidongxu-microsoft** [comment]
@picpromusic
Thanks for the check.

However 2020-08-01-preview is never at roleDefinitions.

I know it is confusing. Microsoft.Authorization uses a mix of api-version for different resource types.

When the title say 2020-10-01, it actually means tag=package-2020-10-01
https://github.com/Azure/azure-rest-api-specs/tree/main/specification/authorization/resource-manager#tag-package-2020-10-01
And if you see the JSON within that tag, roleDefinitions is actually the last stable version of `2015-07-01`

And for 2020-08-01-preview, it is this tag
https://github.com/Azure/azure-rest-api-specs/tree/main/specification/authorization/resource-manager#tag-package-2020-08-01-preview
And roleDefinitions is at `2018-01-01-preview`, which is among the supported api-versions on the error message.

Do let me know which API does not work in 2.14.0. Usually this is service bug, instead of really deprecation of api-version. Deprecation requires lots of approvals so it is not easy, while it is quite easy for service forgetting registering the version on ARM.

**Comment by picpromusic** [comment]
@weidongxu-microsoft 
Unfortunatly i am actually not able to debug it again. But if I remember correctly we got an problem based on https://github.com/Azure/azure-sdk-for-java/blob/0fe5b264ee4d09336713e19ca8046efec0260b33/sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/implementation/RoleAssignmentsClientImpl.java#L1706

If i see it right in https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/implementation/RoleAssignmentsClientImpl.java#L1789
this would be now 2020-10-01-preview which is also not in the list of accepted api-version (based on the error-message). Unfortunatly i haven't the time to do a pre-release and check if it would work with the implementation in main-branch.

**Comment by weidongxu-microsoft** [comment]
I talk about roleDefinition because the error message you gave me above is **roleDefinition**
```
Status code 400, "{"error":{"code":"NoRegisteredProviderFound","message":"No registered resource provider found for location 'westeurope' and API version '2020-08-01-preview' for type 'roleDefinitions'. The supported api-versions are '2014-04-01-preview, 2014-07-01-preview, 2014-10-01-preview, 2015-05-01-preview, 2015-06-01, 2015-07-01, 2016-07-01, 2017-05-01, 2017-09-01, 2018-01-01-preview, 2018-07-01'. The supported locations are ''."}}"
```

And this is the error message on **roleAssignment**, if I use a unsupported api-version.
See it contains both 2020-08-01-preview and 2020-10-01-preview
```
{
  "error" : {
    "code" : "InvalidResourceType",
    "message" : "The resource type 'roleAssignments' could not be found in the namespace 'Microsoft.Authorization' for api version '2020-07-01-preview'. The supported api-versions are '2014-04-01-preview,2014-07-01-preview,2014-10-01-preview,2015-05-01-preview,2015-06-01,2015-07-01,2016-07-01,2017-05-01,2017-09-01,2017-10-01-preview,2018-01-01-preview,2018-07-01,2018-09-01-preview,2018-12-01-preview,2019-04-01-preview,2020-03-01-preview,2020-04-01-preview,2020-08-01-preview,2020-10-01-preview,2021-04-01-preview,2022-01-01-preview'."
  }
}
```

I cannot continue this without correct information. I will drop the investigation.

**Comment by weidongxu-microsoft** [comment]
Maybe you need to check that Microsoft.Authorazation is registered first.

image

File: sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/implementation/RoleDefinitionsClientImpl.java [hunk 1]
```diff
@@ -548,7 +546,7 @@ private Mono<PagedResponse<RoleDefinitionInner>> listSinglePageAsync(String scop
         if (scope == null) {
             return Mono.error(new IllegalArgumentException("Parameter scope is required and cannot be null."));
         }
-        final String apiVersion = "2018-01-01-preview";
+        final String apiVersion = "2015-07-01";
```
[the remark this answers is not in the document]
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/implementation/RoleDefinitionsClientImpl.java:549]: Sure, I will schedule a task to upgrade to "2022-04-01".

As the lib is stable version of e.g. 2.15.0, it was supposed to use stable api-version only.

[the remark this answers is not in the document]
Comment (sachinkumar07) [reply sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/implementation/RoleDefinitionsClientImpl.java:549]: Thank you @weidongxu-microsoft for your prompt action on this. When can we expect the release of 2.21.0?

File: sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/implementation/PermissionImpl.java [hunk 2]
```diff
@@ -26,11 +26,11 @@ public List<String> notActions() {
 
     @Override
     public List<String> dataActions() {
-        return innerModel().dataActions();
+        return innerModel().actions();
```
Comment (sachinkumar07) [inline sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/implementation/PermissionImpl.java:29]: @weidongxu-microsoft dataActions and notDataActions are incorrectly set to actions and notActions.

**Pull request merged**